crates/primitives/src/bridge.rs

```diff
@@ -142,14 +139,28 @@ impl Musig2PartialSig {
 
 impl BorshSerialize for Musig2PartialSig {
     fn serialize<W: std::io::Write>(&self, writer: &mut W) -> std::io::Result<()> {
-        let serialized = self.0.serialize();
+        let sig_bytes = self.0.serialize();
+
+        borsh::BorshSerialize::serialize(&(sig_bytes.len() as u32), writer)?;
 
-        writer.write_all(&serialized)
+        writer.write_all(&sig_bytes)
     }
 }
 
 impl BorshDeserialize for Musig2PartialSig {
     fn deserialize_reader<R: std::io::Read>(reader: &mut R) -> std::io::Result<Self> {
+        let len = u32::deserialize_reader(reader)? as usize;
+
+        if len != MUSIG2_PARTIAL_SIG_SIZE {
+            return Err(std::io::Error::new(
+                std::io::ErrorKind::InvalidData,
+                format!(
+                    "Invalid Musig2PartialSig size, expected: {}, got: {}",
+                    MUSIG2_PARTIAL_SIG_SIZE, len
+                ),
+            ));
```

Comment:
This length tag is redundant if we have a `MUSIG2_PARTIAL_SIG_SIZE`.  If we want to we can fail to serialize if the partial sig is of a different length, but isn't it as `[u8; 64]` or something?  That check should never fail.  AIUI this should just be a passthrough impl onto the inner byte array.

Reply:
Done in https://github.com/alpenlabs/strata/commit/19c2bdb8c54554795da7a9404abd544bfe948ff2.